h2o-hadoop/h2o-mapreduce-generic/src/main/java/water/hadoop/h2odriver.java

```diff
@@ -1075,6 +1131,43 @@ private void addMapperConf(Configuration conf, String name, String value, byte[]
     mapperConfLength++;
   }
 
+  private static ServerSocket bindCallbackSocket() throws IOException {
+    Exception ex = null;
+    int permissionExceptionCount = 0; // try to detect likely unintended range specifications
+    // eg.: running with non-root user & setting range 100-1100 (the effective range would be 1024-1100 = not what user wanted)
+    ServerSocket result = null;
+    for (int p = driverCallbackPortRange.from; (result == null) && (p <= driverCallbackPortRange.to); p++) {
+      ServerSocket ss = new ServerSocket();
+      ss.setReuseAddress(true);
+      InetSocketAddress sa = new InetSocketAddress(driverCallbackIp, p);
+      try {
+        int backlog = Math.max(50, numNodes * 3); // minimum 50 (bind's default) or numNodes * 3 (safety constant, arbitrary)
```

Comment:
I wouldn't expect issues. It is not usual to set the value of the driver port, the typical use case is to leave to 0 (system decides). This means we were setting the backlog argument to 0 in which case the bind(..) overrides the value with the default:

        if (backlog < 1)
          backlog = 50;

That makes me believe that with this change we won't make things worse, we could theoretically improve it if someone had a giant cluster of H2O's (thousands... probably unrealistic).